src/lib/utils.h

```diff
@@ -32,6 +32,15 @@ static inline int min(size_t a, size_t b) {
     return a < b ? a : b;
 }
 
+/*
+ * Pads the buffer 'buf' with blanks to the desired 'buf_len'.
+ * The trailing '\0' byte is removed, so the buffer is NOT null terminared afterwards!
+ * */
+static inline void str_pad(unsigned char * buf, size_t buf_len) {
+    size_t str_len = strlen((char *)buf);
+    memset(buf + str_len, ' ', buf_len - str_len);
```

Comment:
Wouldn't the existing string need to be moved to the end too?

Reply:
all other pkcs11 strings we have are left aligned and padded to the right with blanks.
The standard does not really specify if left or right aligned.
We can  of course change it, but then we have to change str_padded_copy too.

Reply:
https://lists.oasis-open.org/archives/pkcs11/201406/msg00010.html 
If I read this correctly, they must be left aligned and padded to the right

Reply:
before it was a static structure with no additional writes, thus making it thread safe, now it's not thread safe.

Reply:
I see.
Remove the static to make it a local variable, which is thread safe again.?

Reply:
Yep

Reply:
Done&updated